apps/vaporgui/VizWinMgr.cpp

```diff
@@ -345,11 +345,24 @@ void VizWinMgr::_vizAboutToDisappear(string vizName)  {
 
 
 void VizWinMgr::Update(bool fast){
-
+    // Certain actions queue multiple renders within Qt's event queue (e.g. changing the
```

Comment:
The way we have it set up, the render event processing is always done asynchronously from our perspective but synchronously from Qt's perspective. The can be changed to be synchronously from our perspective too but I don't know if that would be a good idea. Qt processes render events one at a time from the event loop, however the event loop can be triggered recursively when updating the progress bar, so if there are multiple render events queued, render can be called recursively. There is no method of preventing this that Qt exposes through its API AFAIK, but this check here prevents any issues. My other PR actually prevents multiple render events from being queued, however I think it would be a good idea to keep this check anyway as a failsafe.